src/Pact/Native/Ops.hs

```diff
@@ -569,5 +569,9 @@ shiftDef =  defRNative "shift" go
   <> "i.e. they fill the top bits with 1 if the x is negative and with 0 otherwise."
   )
   where
-    go _ [TLitInteger x,TLitInteger y] = return $ toTerm $ shift x (fromIntegral y)
+    go i [TLiteral (LInteger x) xi,TLiteral (LInteger y) yi]
+      | y > 0 = do
+          z <- powImpl i [TLiteral (LInteger 2) yi, TLiteral (LInteger y) yi]
```

Comment:
`powImpl` LGTM, but do we want to make it a pure function so we can avoid liften arguments into Terms just to unpack them again? It seems like we could pull that out and avoid some cruft, but it's up to you. Branching looks good on `pow` :+1:  from me.

Reply:
I could factor a version of `intPow` out of `powImpl`, that always succeeds because it only acts on powers > 0, but I'm not sure it's worth the effort for how often shift is actually used on chain (which is zero, as of now).

Reply:
Fair enough - let's punt til it's useful